File: sdk/c/src/bare_metal.c
#include "hitechcloud/bare_metal.h"
#include <stdlib.h>


/* GET /api/service/{id}/reinstall/templates - List OS templates and recipes */
char *hc_bare_metal_list_reinstall_templates(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/reinstall/templates", NULL);
}

/* POST /api/service/{id}/diag/cancel - Cancel diagnostics operation */
char *hc_bare_metal_cancel_diagnostics(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/diag/cancel", NULL);
}

/* GET /api/service/{id}/diag/templates - List diagnostics templates */
char *hc_bare_metal_list_diagnostics_templates(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/diag/templates", NULL);
}

/* GET /api/service/{id}/diag - Show diagnostic operation status */
char *hc_bare_metal_get_diagnostics_status(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/diag", NULL);
}

/* POST /api/service/{id}/diag - Start new diagnostic operation */
char *hc_bare_metal_run_diagnostics(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/diag", NULL);
}

/* GET /api/service/{id}/rescue/templates - List rescue templates */
char *hc_bare_metal_list_rescue_templates(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/rescue/templates", NULL);
}

/* GET /api/service/{id}/rescue - Show rescue operation status */
char *hc_bare_metal_get_rescue_status(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/rescue", NULL);
}

/* POST /api/service/{id}/rescue - Start rescue operation */
char *hc_bare_metal_rescue(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/rescue", NULL);
}

/* POST /api/service/{id}/rescue/cancel - Cancel rescue operation */
char *hc_bare_metal_cancel_rescue(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/rescue/cancel", NULL);
}

/* GET /api/service/{id}/info - Get server hardware and OS details */
char *hc_bare_metal_get_server_info(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/info", NULL);
}

/* POST /api/service/{id}/hostname - Update server hostname */
char *hc_bare_metal_update_hostname(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/hostname", NULL);
}

/* GET /api/service/{id}/ips - List server IPs */
char *hc_bare_metal_list_ips(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/ips", NULL);
}

/* POST /api/service/{id}/ips - Add new IP to server */
char *hc_bare_metal_add_ip(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/ips", NULL);
}

/* GET /api/service/{id}/vlans - List available VLANs */
char *hc_bare_metal_list_vlans(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/vlans", NULL);
}

/* GET /api/service/{id}/ips/{ip} - Get IP details */
char *hc_bare_metal_get_ip_details(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/ips/{ip}", NULL);
}

/* PUT /api/service/{id}/ips/{ip} - Edit IP */
char *hc_bare_metal_edit_ip(hc_http_client_t *client) {
    return hc_put(client, "/api/service/{id}/ips/{ip}", NULL);
}

/* GET /api/service/{id}/reinstall - Return installation options */
char *hc_bare_metal_get_reinstall_details(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/reinstall", NULL);
}

/* POST /api/service/{id}/reinstall - Reinstall server */
char *hc_bare_metal_reinstall(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/reinstall", NULL);
}

/* GET /api/service/{id}/status - Return server power status */
char *hc_bare_metal_get_status(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/status", NULL);
}

/* POST /api/service/{id}/power/{action} - Power action (start/stop/reboot/reset/shutdown) */
char *hc_bare_metal_power_action(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/power/{action}", NULL);
}

/* GET /api/serverstock - List servers in stock */
char *hc_bare_metal_list_servers_in_stock(hc_http_client_t *client) {
    return hc_get(client, "/api/serverstock", NULL);
}

File: sdk/r/R/bare_metal.R
#' Bare Metal & Colocation: reinstall, diagnostics, rescue, server info, IPs, VLANs, power
#' @param http HttpClient object
#' @return Resource object
#' @export
BareMetalResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "BareMetalResource"
  env
}

#' GET /api/service/{id}/reinstall/templates - List OS templates and recipes
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listReinstallTemplates <- function(self, ...) {
  self$http$get("/api/service/{id}/reinstall/templates", list(...))
}

#' POST /api/service/{id}/diag/cancel - Cancel diagnostics operation
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
cancelDiagnostics <- function(self, ...) {
  self$http$post("/api/service/{id}/diag/cancel", list(...))
}

#' GET /api/service/{id}/diag/templates - List diagnostics templates
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listDiagnosticsTemplates <- function(self, ...) {
  self$http$get("/api/service/{id}/diag/templates", list(...))
}

#' GET /api/service/{id}/diag - Show diagnostic operation status
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getDiagnosticsStatus <- function(self, ...) {
  self$http$get("/api/service/{id}/diag", list(...))
}

#' POST /api/service/{id}/diag - Start new diagnostic operation
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
runDiagnostics <- function(self, ...) {
  self$http$post("/api/service/{id}/diag", list(...))
}

#' GET /api/service/{id}/rescue/templates - List rescue templates
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listRescueTemplates <- function(self, ...) {
  self$http$get("/api/service/{id}/rescue/templates", list(...))
}

#' GET /api/service/{id}/rescue - Show rescue operation status
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getRescueStatus <- function(self, ...) {
  self$http$get("/api/service/{id}/rescue", list(...))
}

#' POST /api/service/{id}/rescue - Start rescue operation
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
rescue <- function(self, ...) {
  self$http$post("/api/service/{id}/rescue", list(...))
}

#' POST /api/service/{id}/rescue/cancel - Cancel rescue operation
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
cancelRescue <- function(self, ...) {
  self$http$post("/api/service/{id}/rescue/cancel", list(...))
}

#' GET /api/service/{id}/info - Get server hardware and OS details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getServerInfo <- function(self, ...) {
  self$http$get("/api/service/{id}/info", list(...))
}

#' POST /api/service/{id}/hostname - Update server hostname
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
updateHostname <- function(self, ...) {
  self$http$post("/api/service/{id}/hostname", list(...))
}

#' GET /api/service/{id}/ips - List server IPs
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listIps <- function(self, ...) {
  self$http$get("/api/service/{id}/ips", list(...))
}

#' POST /api/service/{id}/ips - Add new IP to server
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
addIp <- function(self, ...) {
  self$http$post("/api/service/{id}/ips", list(...))
}

#' GET /api/service/{id}/vlans - List available VLANs
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listVlans <- function(self, ...) {
  self$http$get("/api/service/{id}/vlans", list(...))
}

#' GET /api/service/{id}/ips/{ip} - Get IP details
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getIpDetails <- function(self, ...) {
  self$http$get("/api/service/{id}/ips/{ip}", list(...))
}

#' PUT /api/service/{id}/ips/{ip} - Edit IP
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
editIp <- function(self, ...) {
  self$http$put("/api/service/{id}/ips/{ip}", list(...))
}

#' GET /api/service/{id}/reinstall - Return installation options
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getReinstallDetails <- function(self, ...) {
  self$http$get("/api/service/{id}/reinstall", list(...))
}

#' POST /api/service/{id}/reinstall - Reinstall server
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
reinstall <- function(self, ...) {
  self$http$post("/api/service/{id}/reinstall", list(...))
}

#' GET /api/service/{id}/status - Return server power status
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getStatus <- function(self, ...) {
  self$http$get("/api/service/{id}/status", list(...))
}

#' POST /api/service/{id}/power/{action} - Power action (start/stop/reboot/reset/shutdown)
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
powerAction <- function(self, ...) {
  self$http$post("/api/service/{id}/power/{action}", list(...))
}

#' GET /api/serverstock - List servers in stock
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listServersInStock <- function(self, ...) {
  self$http$get("/api/serverstock", list(...))
}

